### PR TITLE
use registerIf to disable metrics in xds cache

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/model/xds_cache.go [hunk 1]
```diff
@@ -30,11 +30,15 @@ import (
...
+var enableStats = func() bool {
+	return features.EnableXDSCacheMetrics
+}
+
 func init() {
-	monitoring.MustRegister(xdsCacheReads)
-	monitoring.MustRegister(xdsCacheEvictions)
-	monitoring.MustRegister(xdsCacheSize)
-	monitoring.MustRegister(dependentConfigSize)
+	xdsCacheReads = monitoring.RegisterIf(xdsCacheReads, enableStats)
```
Comment (howardjohn) [inline pilot/pkg/model/xds_cache.go:38]: should we just wrap L45 instead of using init?

File: pilot/pkg/model/xds_cache.go [hunk 2]
```diff
@@ -30,35 +30,28 @@ import (
 	"istio.io/pkg/monitoring"
 )
 
-func init() {
-	monitoring.MustRegister(xdsCacheReads)
-	monitoring.MustRegister(xdsCacheEvictions)
-	monitoring.MustRegister(xdsCacheSize)
-	monitoring.MustRegister(dependentConfigSize)
+var enableStats = func() bool {
+	return features.EnableXDSCacheMetrics
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/xds_cache.go:34]: no need a func wrapper

Comment (ramaraochavali) [reply pilot/pkg/model/xds_cache.go:34]: Sorry. I did not understand. Can you please eloborate?

Comment (hzxuzhonghu) [reply pilot/pkg/model/xds_cache.go:34]: I mean

var enableStats = features.EnableXDSCacheMetrics

Comment (ramaraochavali) [reply pilot/pkg/model/xds_cache.go:34]: But RegisterIf takes a func that returns bool?

Comment (hzxuzhonghu) [reply pilot/pkg/model/xds_cache.go:34]: OK, but i doubt it is not necessary in this case.

Comment (ramaraochavali) [reply pilot/pkg/model/xds_cache.go:34]: Are you suggesting to rework RegisterIf function?

Comment (hzxuzhonghu) [reply pilot/pkg/model/xds_cache.go:34]: I am not sure, just feel it makes things a little complex with cb

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**